### PR TITLE
update limits and codes

### LINKED ISSUE #114: update limits and codes internal data
to reflect new limits being used by ShinyRems app (i.e. it is an expanded version of the current limits and codes - but not the full guidelines sheet on DataBC

### PR DESCRIPTION
**Pull request by sebdalgarno**
I have updated the limits and codes table in wqbc and have had to update some tests as well to test new values (from adjusted Limits) and removed variables (Copper Total, Arsenic Total, etc).

The updated limits are a simplified version of  the new guidelines spreadsheet developed by Angeline Tillmans and co. Only limits with Use = "Freshwater Life" and Media = "Water" have been kept so as to require minimal changes in the actual calc_limits function code.

Note more limits (for different users/media etc.) might be added next year but for now we wanted to provide the basic functionality for some new/expanded limits.

To see the whole new guidelines data:
```
limits_new <- bcdata::bcdc_get_data(record = "85d3990a-ec0a-4436-8ebd-150de3ba0747")
```

fixes #114

### CODE REVIEW

**Comment by ateucher** [comment]
One question: why not just fully replace the old limits.csv with the processed data from the catalogue dataset, rather than merging/conditionally replacing parts?

**Comment by sebdalgarno** [comment]
none of the old limits have been moved into the new limits table. The section where a series of tmp objects are created (e.g. `tmp <- anti_join(limits, limits_new, c("EMS_Code", "Term"))` is simply me confirming to myself (and anyone interested) that there is a reason for each old limit missing in the new limit table.

I have bound any missing old codes to the new codes table, and these  could probably be removed (ie. just create new codes table from the new limits table)

Does that answer your question?

**Comment by sebdalgarno** [comment]
I'm going to get Angeline Tillmanns to update the guidelines spreadsheet on DataBC as it is getting to a point of stability now. I will then update the limits data again and clean up the 'limits-update.R' script. I think this will help to remove some of the cleaning required and other things in consideration of your comment: "One question: why not just fully replace the old limits.csv with the processed data from the catalogue dataset, rather than merging/conditionally replacing parts?"

**Comment by ateucher** [comment]
Makes sense @sebdalgarno - would you prefer I merge this now so you can move along with your other plans, then you can submit a new PR once the guidelines are updated in the data catalogue? Or do you want to hold off and do it all in one go?

**Comment by sebdalgarno** [comment]
thanks @ateucher. Angeline just updated the guidelines. I will push the updates this morning and you can merge in one go.

**Comment by sebdalgarno** [comment]
OK @ateucher ready for final review and merge

File: data-raw/limits-update.R [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+
+
+### deal with hardness equations (only include Hardness Total when both Hardness Total and Hardnes Dissolved)
+modified <- limits_new$Condition[which(stringr::str_detect(limits_new$Condition, "EMS_0107"))] %>%
+  stringr::str_split_fixed("\\|", 2)
+modified <- modified[, 1]
+limits_new$Condition[which(stringr::str_detect(limits_new$Condition, "EMS_0107"))] <- modified
+
+### replace Aluminum with Aluminium
+limits_new$Variable %<>%
```
Comment (ateucher) [inline data-raw/limits-update.R]: I don't think `magrittr` is loaded anywhere for `%<>%`

File: data-raw/limits-update.R [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+  stringr::str_replace_all("Aluminum", "Aluminium")
+
+### create new codes table and grab EC_Code from old table where possible
+ec_codes <- select(codes, Code, EC_Code) %>%
+  filter(Code %in% unique(limits$EMS_Code), !is.na(EC_Code))
```
[the remark this answers is not in the document]
Comment (sebdalgarno) [reply data-raw/limits-update.R]: yes...making that change

File: data-raw/limits-update.R [hunk 3]
```diff
@@ -0,0 +1,146 @@
+library(dplyr)
+
+load("data/codes.rda")
+load("data/limits.rda")
```
[the remark this answers is not in the document]
Comment (sebdalgarno) [reply data-raw/limits-update.R]: we still need the old codes data (for getting EC_Code) but I will remove the limits

File: data-raw/limits-update.R [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+  dplyr::ungroup()
+
+### ensure that no duplicates
+expect_true(all(limits_new %>%
+                  dplyr::group_by(EMS_Code, Use, Term, Condition) %>%
+                  dplyr::mutate(n = dplyr::n()) %>%
+                  dplyr::ungroup() %>%
+                  dplyr::pull(n) == 1))
+
+# limits <- limits %>%
```
[the remark this answers is not in the document]
Comment (sebdalgarno) [reply data-raw/limits-update.R]: removing

File: data-raw/limits-update.R [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+  ungroup()
+
+### remove duplicates
+limits_new <- limits_new %>%
+  dplyr::group_by(EMS_Code, Use, Term, Condition) %>%
+  dplyr::filter(dplyr::n() == 1) %>%
+  dplyr::ungroup()
+
+### ensure that no duplicates
+expect_true(all(limits_new %>%
```
[the remark this answers is not in the document]
Comment (sebdalgarno) [reply data-raw/limits-update.R]: changing to stopifnot

File: data-raw/limits-update.R [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+#   filter(!(EMS_Code == "EMS_AK_T"))
+#
+# ### confirm that all cases have been dealt with
+# expect_identical(nrow(tmp), 0L)
+
+
+### deal with hardness equations (only include Hardness Total when both Hardness Total and Hardnes Dissolved)
+modified <- limits_new$Condition[which(stringr::str_detect(limits_new$Condition, "EMS_0107"))] %>%
+  stringr::str_split_fixed("\\|", 2)
+modified <- modified[, 1]
```
[the remark this answers is not in the document]
Comment (sebdalgarno) [reply data-raw/limits-update.R]: I can be for the current data but potentially not in future. I have made this more robust by grabbing the side of the | that contains EMS_0107

File: data-raw/limits-update.R [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+limits_new$Condition[which(stringr::str_detect(limits_new$Condition, "EMS_0107"))] <- modified
+
+### replace Aluminum with Aluminium
+limits_new$Variable %<>%
+  stringr::str_replace_all("Aluminum", "Aluminium")
```
[the remark this answers is not in the document]
Comment (sebdalgarno) [reply data-raw/limits-update.R]: I'm removing references to Aluminium in favour of Aluminum

File: data-raw/limits-update.R (hunk 1 above)
Comment (sebdalgarno) [reply data-raw/limits-update.R]: removed

**Comment by sebdalgarno** [comment]
OK @ateucher thanks for your patience on this one and for the good review. I've now pushed changes that address all of your comments/suggestions above, including renaming limits-update.R to limits.R

**Comment by ateucher** [comment]
Looking good @sebdalgarno, thanks! Can you now delete `limits-update.R`?

**Comment by ateucher** [comment]
Thanks! I'm going to merge this, then push up a couple of local changes I made to some of the other `data-raw` scripts.

**Comment by sebdalgarno** [comment]
OK thanks Andy! that wraps up all the PRs I was planning on making to wqbc and rems. I think @joethorley is working on a PR for wqbc (on censored data)

**Pull request merged**